lib/galaxy/schema/tools.py

```diff
@@ -0,0 +1,167 @@
+from galaxy.schema.jobs import (
+    EncodedJobDetails,
+    ShowFullJobResponse,
+)
+from galaxy.schema.schema import (
+    HDACustom,
+    HDADetailed,
+    HDASummary,
+    HDCADetailed,
+    HDCASummary,
+    JobSummary,
+    Model,
+)
+
+ToolOutputName = Annotated[
+    str,
+    Field(
+        ...,
+        title="Output Name",
+        description="The name of the tool output",
+    ),
+]
+
+
+class ExecuteToolPayload(Model):
+    tool_id: Optional[str] = Field(
+        default=None,
+        title="Tool ID",
+        description="The ID of the tool to execute.",
+    )
+    tool_uuid: Optional[UUID4] = Field(
+        default=None,
+        title="Tool UUID",
+        description="The UUID of the tool to execute.",
+    )
+    action: Optional[str] = Field(
+        default=None,
+        title="Action",
+        description="The action to perform",
+    )
+    tool_version: Optional[str] = Field(
+        default=None,
+        title="Tool Version",
+        description="The version of the tool",
+    )
+    history_id: Optional[DecodedDatabaseIdField] = Field(
+        default=None,
+        title="History ID",
+        description="The ID of the history",
+    )
+
+    @field_validator("inputs", mode="before", check_fields=False)
+    @classmethod
+    def inputs_string_to_json(cls, v):
+        if isinstance(v, str):
+            return json.loads(v)
+        return v
+
+    inputs: Dict[str, Any] = Field(
+        default={},
+        title="Inputs",
+        description="The input data",
+    )
+    use_cached_job: Optional[bool] = Field(
+        default=False,
+        title="Use Cached Job",
+        description="Flag indicating whether to use a cached job",
+    )
+    preferred_object_store_id: Optional[str] = Field(
+        default=None,
+        title="Preferred Object Store ID",
+        description="The ID of the preferred object store",
+    )
+    input_format: Optional[str] = Field(
+        default="legacy",
+        title="Input Format",
+        description="The format of the input",
+    )
+    data_manager_mode: Optional[str] = Field(
+        default=None,
+        title="Data Manager Mode",
+        description="The mode of the data manager",
+    )
+    send_email_notification: Optional[bool] = Field(
+        default=None,
+        title="Send Email Notification",
+        description="Flag indicating whether to send email notification",
+    )
+    upload_type: Optional[str] = Field(
+        default=None,
+        title="Upload Type",
+        description="TODO",
+    )
+    model_config = ConfigDict(extra="allow")
+
+
+class HDACustomWithOutputName(HDACustom):
+    output_name: ToolOutputName
+
+
+class HDADetailedWithOutputName(HDADetailed):
+    output_name: ToolOutputName
+
+
+class HDASummaryWithOutputName(HDASummary):
+    output_name: ToolOutputName
+
+
+AnyHDAWithOutputName = Union[HDACustomWithOutputName, HDADetailedWithOutputName, HDASummaryWithOutputName]
```

Comment:
I'm not sure if having multiple models here is needed. It seems there is only one way to serialize the output datasets:

https://github.com/galaxyproject/galaxy/blob/bc248efe2aad464f5df12d8bdd864c8be146d7a5/lib/galaxy/webapps/galaxy/services/tools.py#L210

So probably having something like the following will be enough:

```python
class HDAToolOutput(HDACustom):
    output_name: ToolOutputName
```

The same can be applied to `output_collections`, `jobs`, and `implicit_collections`.  Have a closer look at what gets serialized in `_handle_inputs_output_to_api_response` and use just one model for those. Using multiple models is useful only when you have a way to specify different serialization parameters, but this endpoint seems to return a fixed set of attributes for each model.

Reply:
It's also fine if you just return the widest set (which I think is HDACustomWithOutputName ?) ... the client should ideally not look at the job response as we'll hopefully make that async in the future. Right now all we use this for is this little banner message saying "N outputs were created", and if it's an interactive tool we show the entry point.